### ISSUE TITLE
Changing context in a Stache template breaks the data helper

### ISSUE THREAD

**Issue by dylanrtt** [bitovi/canjs#1619]
In the migration from EJS to Mustache, we lost the ability to explicitly declare which data was used with the data helper. For example, in EJS you could use:

```<div (el) -> can.data(el, 'todo', todo)></div>```

In some cases, we rewrote the JS to expect the data to be the entire context, but in others, we worked around this limitation by changing the context in the template surrounding the usage of the data helper. For example:

```<div {{#todo}} {{ data "todo" }} {{/todo}}></div>```

Unfortunately, while upgrading to Stache, this broke due to an apparent bug that seems to put JS code into the DOM as exhibited by this fiddle.

http://jsfiddle.net/qYdwR/2476/

So far, this seems to only be a problem with ```{{#key}}{/key}}``` and not with ```{{#each key}}{{/each}}``` which also changes context. A similar issue also occurs with using ```#if``` to wrap around the data helper instead.

**Comment by daffl** [comment]
The result is indeed odd but I think the easiest solution for now is to pass the context you want to store in the data as the second argument:

```html
<div id="test" {{ data "todo" todo }}></div>
```

```js
$('#test').data('todo') // -> todo
```

**Comment by dylanrtt** [comment]
Oh, I didn't realize that was supported. I could have sworn I tested this before and that it did not work... but it seems to be working fine now.

You should add that to the docs so people know it's available.

**Comment by justinbmeyer** [comment]
@daffl Bug? Or docs?

**Comment by daffl** [comment]
I would say docs for this and @marshallswain already started that in #1640.

**Comment by alexisabril** [comment]
Closed by b7c8769b26c4b56e866f5d5804c353693bd55ad3
